src/metabase/sync/sync_metadata/fields.clj

```diff
@@ -104,19 +112,6 @@
   ([database :- i/DatabaseInstance
     table    :- i/TableInstance]
    (sync-util/with-error-handling (format "Error syncing Fields for Table ''%s''" (sync-util/name-for-logging table))
-     (let [db-metadata (fields.our-metadata/db-metadata database table)]
+     (let [db-metadata  (fetch-metadata/table-fields-metadata database table)]
        {:total-fields   (count db-metadata)
         :updated-fields (sync-and-update! table db-metadata)}))))
-
-(mu/defn sync-fields! :- [:maybe
-                          [:map
-                           [:updated-fields ms/IntGreaterThanOrEqualToZero]
-                           [:total-fields   ms/IntGreaterThanOrEqualToZero]]]
-  "Sync the Fields in the Metabase application database for all the Tables in a `database`."
-  [database :- i/DatabaseInstance]
-  (if (driver/database-supports? (driver.u/database->driver database) :describe-fields database)
```

Comment:
This branching is what this PR is all about. It's been moved to the `fetch-metadata/fields-metadata` function [here](https://github.com/metabase/metabase/pull/41588/files#diff-66e35cc247f06dac20016cb194a545e520bb1a89de30ad2d1bcf7bb2011b008aR72-R76).